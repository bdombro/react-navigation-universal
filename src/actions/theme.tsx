import {ActionTypes} from "./actionTypes";
// import {Theme} from "../reducers";

export interface ToggleThemeAction {
    type: ActionTypes.toggleTheme;
    // payload: Theme
}

export const toggleTheme = () => ({
    type: ActionTypes.toggleTheme,
    // payload: {},
});
