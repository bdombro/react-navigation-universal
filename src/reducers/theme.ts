import { Action, ActionTypes } from '../actions';
import { Theme } from './';
import {ThemeConfig} from "../config/Theme.config";
export const themeReducer = (state: Theme = ThemeConfig.light, action: Action) => {
  switch (action.type) {
    case ActionTypes.toggleTheme:
      return state.dark ? ThemeConfig.light : ThemeConfig.dark;
    default:
      return state;
  }
};
