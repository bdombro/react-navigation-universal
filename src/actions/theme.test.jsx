import {toggleTheme} from "./theme";

describe('actions', () => {
    it('should create an action to toggle the theme', () => {
        expect(toggleTheme()).toMatchSnapshot();
    });
});
