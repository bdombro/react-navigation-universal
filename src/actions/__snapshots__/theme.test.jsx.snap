// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create an action to toggle the theme 1`] = `
Object {
  "type": 2,
}
`;
